File: frontend/src/model/Ingredient.ts
export type Ingredient = {
    id: string,
    name: string,
    quantity: string,
    unit: string,
    productCategory?: ProductCategory | string,
    isInShoppingList?: boolean,
}
export type NewItem = {
    name: string,
    quantity: string,
    unit: string,
    isInShoppingList: boolean
}



export enum ProductCategory {
    BREAD_BAKERY = "BREAD_BAKERY",
    BEVERAGES = "BEVERAGES",
    CANNED_JARED_GOODS = "CANNED_JARED_GOODS",
    DAIRY = "DAIRY",
    PRODUCE = "PRODUCE",
    DRY_BACKING_GOODS = "DRY_BACKING_GOODS",
    FROZEN_FOODS = "FROZEN_FOODS",
    MEAT = "MEAT",
    PERSONAL_CARE = "PERSONAL_CARE",
    OTHERS = "OTHERS"
}

File: backend/src/main/java/com/github/licetrosales/backend/model/Ingredient.java
package com.github.licetrosales.backend.model;

import lombok.With;
import org.springframework.data.annotation.Id;


@With
public record Ingredient(
        @Id
        String id,
        String name,
        String quantity,
        String unit,
        boolean isInShoppingList,
        ProductCategory productCategory
) {
    Ingredient(
            String name,
            String quantity,
            String unit,
            Boolean isInShoppingList
    ) {
        this(null, name, quantity, unit, false, null);
    }
}